src/posix/termios.rs

```diff
@@ -163,6 +163,20 @@ pub(crate) fn set_parity(termios: &mut Termios, parity: Parity) {
             termios.c_iflag |= libc::INPCK;
             termios.c_iflag &= !libc::IGNPAR;
         }
+        Parity::Mark => {
+            termios.c_cflag |= libc::PARODD;
+            termios.c_cflag |= libc::PARENB;
+            termios.c_iflag |= libc::INPCK;
+            termios.c_iflag &= !libc::IGNPAR;
+            termios.c_iflag |= libc::CMSPAR;
```

Comment:
So apparently this only works on Linux. What should it do on other targets? Do the same as odd/even, panic, or something else?

Reply:
Thank you for starting this PR! Thank you especially for looking at Windows as well! :smile:

Mark and space parity is not supported on all platforms of our POSIX backend. As far as i've looked into it, its controlled by the flag `CMSPAR` where present. I would provide support for mark and space parity only on these ones and make `Parity::Mark` and `Parity::Space` conditionally available via `#[cfg(...)]`.

Reply:
I've made the change, but I think it's problematic for a few reasons:

1. If someone wanted to match on the `Parity` (e.g., to provide an icon, color, localized string etc for each variant), they would have to add the same verbose `cfg` everywhere, and they would have to look at the library source to get it right.
2. If another existing target gets support for those variants in the future in one way or another, that would be a new breaking change for the library. Unless the enum is marked `#[non_exhaustive]`, but then you'd have to add a wildcard arm to every match statement, which is annoying and makes the code more brittle.

So maybe it would be better to just make the variants available unconditionally, and document which targets support them. I'm not sure how other libraries like QSerialPort and pySerial handle this. I can try to look into it.

Reply:
Maybe there could be a `supported()` method for the enum, returning false for targets which are not supported, so the user can just filter them out without getting into the details of all the Unix variants. And then the `SerialPortBuilder` could panic when the variant is not supported, instead of invoking unspecified behavior.

Reply:
Actually, it doesn't need to panic at all. The methods already return `Result`, so I can just make `termios::set_parity` return an error with a descriptive message when the platform is not supported, and propagate that to the public methods. The `open` method already does that:

https://github.com/serialport/serialport-rs/blob/c3209197dfbc0c760c843c7905d49b83b99714e3/src/lib.rs#L385-L397

Does this seem reasonable? I'm going to revert the previous commit and try that to see how it looks.

Reply:
Thank you very much for pushing this forward! Well, I think, that I now see what you meant with `supported()`. It's `Parity::supported(&self) -> bool`, right. And at the end this might be the Pareto solution to this issue.

Parity is likely being set by an application, it will be done in few cases and even fewer people will use mark and space parity at all. Compared to `Parity` being `#[non_exhaustive]` and having `Mark` and `Space` conditionally available, it has its strength in a less cluttered API.

Let me sleep over it.